File: gluecodium/src/test/resources/smoke/generic_types/output/android/com/example/smoke/GenericTypesWithBasicTypes.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;
import java.util.Map;
import java.util.Set;
public final class GenericTypesWithBasicTypes extends NativeBase {
    public static final class StructWithGenerics {
        @NonNull
        public List<Short> numbersList;
        @NonNull
        public Map<Short, String> numbersMap;
        @NonNull
        public Set<Short> numbersSet;
        public StructWithGenerics(@NonNull final List<Short> numbersList, @NonNull final Map<Short, String> numbersMap, @NonNull final Set<Short> numbersSet) {
            this.numbersList = numbersList;
            this.numbersMap = numbersMap;
            this.numbersSet = numbersSet;
        }
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected GenericTypesWithBasicTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native List<Integer> methodWithList(@NonNull final List<Integer> input);
    @NonNull
    public native Map<Integer, Boolean> methodWithMap(@NonNull final Map<Integer, Boolean> input);
    @NonNull
    public native Set<Integer> methodWithSet(@NonNull final Set<Integer> input);
    @NonNull
    public native List<String> methodWithListTypeAlias(@NonNull final List<String> input);
    @NonNull
    public native Map<String, String> methodWithMapTypeAlias(@NonNull final Map<String, String> input);
    @NonNull
    public native Set<String> methodWithSetTypeAlias(@NonNull final Set<String> input);
    @NonNull
    public native List<Float> getListProperty();
    public native void setListProperty(@NonNull final List<Float> value);
    @NonNull
    public native Map<Float, Double> getMapProperty();
    public native void setMapProperty(@NonNull final Map<Float, Double> value);
    @NonNull
    public native Set<Float> getSetProperty();
    public native void setSetProperty(@NonNull final Set<Float> value);
}

File: gluecodium/src/test/resources/smoke/generic_types/output/android-kotlin/com/example/smoke/GenericTypesWithBasicTypes.kt
/*

 *
 */

@file:JvmName("GenericTypesWithBasicTypesExtensions")


package com.example.smoke

import com.example.NativeBase

class GenericTypesWithBasicTypes : NativeBase {

    class StructWithGenerics {
        @JvmField var numbersList: List<Short>
        @JvmField var numbersMap: Map<Short, String>
        @JvmField var numbersSet: Set<Short>



        constructor(numbersList: List<Short>, numbersMap: Map<Short, String>, numbersSet: Set<Short>) {
            this.numbersList = numbersList
            this.numbersMap = numbersMap
            this.numbersSet = numbersSet
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun methodWithList(input: List<Int>) : List<Int>


    external fun methodWithMap(input: Map<Int, Boolean>) : Map<Int, Boolean>


    external fun methodWithSet(input: Set<Int>) : Set<Int>


    external fun methodWithListTypeAlias(input: List<String>) : List<String>


    external fun methodWithMapTypeAlias(input: Map<String, String>) : Map<String, String>


    external fun methodWithSetTypeAlias(input: Set<String>) : Set<String>

    var listProperty: List<Float>
        external get
        external set

    var mapProperty: Map<Float, Double>
        external get
        external set

    var setProperty: Set<Float>
        external get
        external set




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


